### PR TITLE
Update UniLoRA MetaMathQA benchmark configs with tuned hyperparameters

### PR DESCRIPTION
**Pull request by peft-jambot**
## Summary

Replaces the default UniLoRA experiment config for the MetaMathQA benchmark with a properly tuned configuration based on the Uni-LoRA paper (arXiv:2506.00799) hyperparameters.

The original config (`llama-3.2-3B-default`) used `r=32`, `theta_d_length=256`, and `lr=1e-4`, which produced **0.15% test accuracy** — essentially random — because 256 trainable parameters with a low learning rate are insufficient for the model to learn.

## New config

| Config | r | θ_d_length | LR | Trainable | Test Acc |
|---|---|---|---|---|---|
| `llama-3.2-3B-r4-td131072-lr0.002` | 4 | 131,072 | 2e-3 | 131K | 41.6% |

The config uses `r=4` (per the paper's Table 9) since a parameter sweep showed rank has negligible impact on accuracy (41.6% vs 37.4% for r=4 vs r=32 at 131K) while directly increasing the frozen projection matrix size and memory consumption.

## Comparison with other PEFT methods on the same benchmark

| Method | Trainable | Test Acc |
|---|---|---|
| UniLoRA (original, broken) | 256 | 0.15% |
| **UniLoRA (r4, td=131K)** | **131K** | **41.6%** |
| VeRA | 129K | 35.4% |
| P-VeRA (r=64) | 122K | 35.6% |
| LoKr (r=32) | 280K | 36.0% |
| PEANUT (r=1) | 229K | 39.7% |
| LoRA (r=32) | 9.2M | 47.8% |

The 131K-parameter config is competitive with VeRA (129K params, 35.4%) while achieving higher accuracy (41.6% vs 35.4%), making it a good parameter-efficiency comparison point.

## Paper alignment

The hyperparameters are sourced from the Uni-LoRA paper's Table 9 (mathematical reasoning experiments): `r=4`, `lr=2e-3`, cosine schedule, warmup ratio 0.02, init `U(-0.02, 0.02)`. The paper reports 68.5% GSM8K accuracy on Mistral-7B with 524K params; we use a smaller 131K subspace for Llama-3.2-3B (smaller model, different benchmark setup).

## Testing

No code changes — only benchmark experiment configuration files. Validated by running the config via HF Jobs on `huggingface/peft-gpu:latest` (a10g-large), completed successfully with the reported accuracy.

AI assistance was used to run the parameter sweep and prepare this PR.

### CODE REVIEW

**Review by githubnemo — CHANGES_REQUESTED** [review]
Let's only use the td13... experiment since it showcases good performance and comparability to the likes of VeRA. Maybe we someday are in need of parameter-efficient instead of compute or memory-efficient methods and then this will be a good comparison.

**Review by githubnemo — APPROVED** [review]
OK for me. @BenjaminBossan here are the results from the runner:

```
$ grep -E 'accuracy|forget|memory' temporary_results/unilora--llama-3.2-3B-r4-td131072-lr0.002--2026-09-10T13-30-57+00-00.json 
    "accelerator_memory_reserved_avg": 14296927331,
    "accelerator_memory_max": 22162702336,
    "accelerator_memory_reserved_99th": 20011024384,  
        "test accuracy": 0.41470811220621684,
        "forgetting": 0.08203792572021484       

$ grep -E 'accuracy|forget|memory' results/lora--llama-3.2-3B-rank32.json 
    "accelerator_memory_reserved_avg": 14397338969,
    "accelerator_memory_max": 22292725760,      
    "accelerator_memory_reserved_99th": 20120076288,
        "test accuracy": 0.4783927217589083,
        "forgetting": 0.04995572566986084

$ grep -E 'accuracy|forget|memory' temporary_results/unilora--llama-3.2-3B-r4-td524288-lr0.002--2026-09-10T13-53-50+00-00.json 
    "accelerator_memory_reserved_avg": 14291697873,
    "accelerator_memory_max": 22194159616,
    "accelerator_memory_reserved_99th": 20008927232,
        "test accuracy": 0.44351781652767247,
        "forgetting": 0.11280357837677002
```

Note that we removed the td5* experiment.

**Review by BenjaminBossan — APPROVED** [review]
LGTM, thx.
